### ISSUE TITLE
Websocket issue Unsubscribe

### ISSUE THREAD

**Issue by BMGDigitalTech** [bchavez/Coinbase.Pro#41]
I dont see unit testing.. so I am guessing no one has used this function because it's not right.

      public void Unsubscribe(Subscription subscription)
      {
         /// issue here
         **subscription.ExtraJson.Add("type", JToken.FromObject(MessageType.Unsubscribe));**
         /// I am unsubscribing.. should this be "subscription.ExtraJson.Remove("type" );"

        /// right now its throwing an exception since that subscription already exists

         var json = JsonConvert.SerializeObject(subscription);

         this.RawSocket.Send(json);
      }

**Comment by BMGDigitalTech** [comment]
Ok.. so my bad that is not the issue.. so after a lot of testing if found the issue.   at 1st I thought that object was just list of the subscription related to the socket. but after testing I found it's not..

1. I was storing the subscription and  just passing that, and since it had type as subscribe already set in the array that's what the issue was.. so I modified Unsubscribe(Subscription subscription), and added 
            if(subscription.ExtraJson.ContainsKey("type"))
            {
                subscription.ExtraJson["type"] = JToken.FromObject(MessageType.Unsubscribe); //  ("type", JToken.FromObject(MessageType.Unsubscribe));
            }
            else
            {
                subscription.ExtraJson.Add("type", JToken.FromObject(MessageType.Unsubscribe));
            } 

this way I could continue to do what I expected.. than it still didn't work..

2. the real issue: on Coinbase.Pro.Models.MessageType there is a space at the end of  Unsubscribe
      [EnumMember(Value = "unsubscribe ")]

I fixed that and it started working..
